client/src/app/_services/template.service.ts

```diff
@@ -9,7 +9,14 @@ export class TemplateService {
     constructor(private http: HttpClient) { }
 
     getAll() {
-        // TODO change return type to Template[]
-        return this.http.get<Template>('/api/services/template');
+        return this.http.get<Template[]>('/api/templates');
+    }
+
+    getById(id: number) {
+        return this.http.get<Template>('/api/templates/' + id);
+    }
+
+    create(template: Template) {
+        return this.http.post('/api/templates', {template: template} );
```

Comment:
You need to send the body like this: `http.post('/api/templates', template)`.

Check the back if you ant to be sure :smile: